src/serializer/functions.js

```diff
@@ -255,29 +255,39 @@ export class Functions {
     return noOpFunc;
   }
 
+  _forEachBindingOfEffects(effects: Effects, func: (binding: Binding) => void): void {
+    let [result, , nestedBindingsToIgnore] = effects.data;
+    for (let [binding] of nestedBindingsToIgnore) {
+      func(binding);
+    }
+    if (result instanceof PossiblyNormalCompletion) {
+      this._forEachBindingOfEffects(result.alternateEffects, func);
+      this._forEachBindingOfEffects(result.consequentEffects, func);
+    }
```

Comment:
It can also have `result.savedEffects`, do we need to visit those?

Reply:
I'm not sure if we should visit them, maybe @hermanventer can shine some light.